### PR TITLE
ci: Added Windows workflow

### LINKED ISSUE #1528: Re-introduce Windows as a build platform
**Author the user story for this feature**

As a developer I want to ensure Windows is supported for flank. That includes tests.

**Describe the solution**
Lets include a Windows instance that can run a flank test suite (Unit tests and gradle clean build) to ensure we test it correctly

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1528

Windows workflow runs and passes

## Checklist

- [x] Added Windows workflow

### CODE REVIEW

File: .github/workflows/windows_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: macOS-workflow
```
Comment (axelzuziak-gogo) [inline .github/workflows/windows_workflow.yml:1]: Am I missing something here?

Comment (piotradamczyk5) [reply .github/workflows/windows_workflow.yml]: I commited bad file, fixed. Thanks

**Review by axelzuziak-gogo — APPROVED** [review]

File: .github/workflows/windows_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: windows-workflow
+
+on:
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    runs-on: windows-latest
```
Comment (Sloox) [inline .github/workflows/windows_workflow.yml:10]: Is there a set windows version? Do we know it? 
Do they update it often?
Windows is notoriously unstable across versions so I want to be sure we are constant.

**Review by Sloox — CHANGES_REQUESTED** [review]

File: .github/workflows/windows_workflow.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-2-gradle-${{ hashFiles('**/*.gradle*') }}
+        restore-keys: |
+          ${{ runner.os }}-2-gradle-
+
+    - name: Gradle clean build
+      uses: eskatos/gradle-command-action@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        HEAD_REF: ${{ github.head_ref }}
```
Comment (Sloox) [inline .github/workflows/windows_workflow.yml:33]: I think you need to add this to get it to be succesful:
```suggestion
        HEAD_REF: ${{ github.head_ref }}
        FLANK_PROJECT_ID: flank-open-source
```
or possibily 
```suggestion
        HEAD_REF: ${{ github.head_ref }}
        GOOGLE_CLOUD_PROJECT: flank-open-source
```

Comment (piotradamczyk5) [reply .github/workflows/windows_workflow.yml:33]: this task run only local unit tests, so I do not think we need to specify project id

File: .github/workflows/windows_workflow.yml (hunk 2 above)
Comment (piotradamczyk5) [reply .github/workflows/windows_workflow.yml:10]: Version is specified at https://github.com/actions/virtual-environments

GitHub team make a version as `-latest` only when it is stable (ex. ubuntu still use 18.04 as latest)

File: .github/workflows/windows_workflow.yml (hunk 3 above)
Comment (Sloox) [reply .github/workflows/windows_workflow.yml:33]: @piotradamczyk5 the run is failing as the project field is null. I assume the project field requires this correct?
https://github.com/Flank/flank/pull/1534/checks?check_run_id=1767430233

Comment (Sloox) [reply .github/workflows/windows_workflow.yml:33]: @piotradamczyk5 I see a lot of errors actually nevermind.

Comment (piotradamczyk5) [reply .github/workflows/windows_workflow.yml:33]: I will check them, they could be related to other things

File: .github/workflows/windows_workflow.yml (hunk 2 above)
Comment (Sloox) [reply .github/workflows/windows_workflow.yml:10]: 👍

File: test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt [hunk 4]
```diff
@@ -15,7 +15,7 @@ fun CommonConfig.createCommonArgs(
     // gcloud
     devices = gcloud::devices.require(),
     resultsBucket = ArgsHelper.createGcsBucket(
-        projectId = flank::project.require(),
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt:18]: Why this change? Project id must never be empty

**Review by pawelpasterz — CHANGES_REQUESTED** [review]

File: test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt [hunk 5]
```diff
@@ -41,7 +41,7 @@ fun CommonConfig.createCommonArgs(
     testTargetsAlwaysRun = flank::testTargetsAlwaysRun.require(),
     runTimeout = flank::runTimeout.require(),
     fullJUnitResult = flank::fullJUnitResult.require(),
-    project = flank::project.require(),
+    project = flank.project.orEmpty(),
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt:44]:
```suggestion
    project = flank::project.require(),
```

File: test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt [hunk 6]
```diff
@@ -15,7 +15,7 @@ fun CommonConfig.createCommonArgs(
     // gcloud
     devices = gcloud::devices.require(),
     resultsBucket = ArgsHelper.createGcsBucket(
-        projectId = flank::project.require(),
+        projectId = flank.project.orEmpty(),
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt:18]:
```suggestion
        projectId = flank::project.require(),
```

File: test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt (hunk 4 above)
Comment (piotradamczyk5) [reply test_runner/src/main/kotlin/ftl/args/CreateCommonArgs.kt]: reverted

File: common/src/main/kotlin/flank/common/Files.kt [hunk 7]
```diff
@@ -13,6 +13,10 @@ val userHome: String by lazy {
     if (isWindows) System.getenv("HOMEPATH") else System.getProperty("user.home")
 }
 
+val appDataDirectory: String by lazy {
+    if (isWindows) System.getenv("APPDATA") else System.getProperty("user.home")
```
Comment (Sloox) [inline common/src/main/kotlin/flank/common/Files.kt:17]: I had a feeling something related to HOMEPATH was going to be the issue. I assume that HOMEPATH you dont have write permissions on the VM'S right?

**Review by Sloox — APPROVED** [review]

File: common/src/main/kotlin/flank/common/Files.kt (hunk 7 above)
Comment (piotradamczyk5) [reply common/src/main/kotlin/flank/common/Files.kt:17]: exactly. User directory is for users 🎉  Just one working day to figure this out

Comment (piotradamczyk5) [reply common/src/main/kotlin/flank/common/Files.kt:17]: and basically this is good practice on Windows to keep app data into app data folder instead of user one
